js/index.js

```diff
@@ -26,13 +26,85 @@ const skillsSection = document.getElementById('skills');
-  // Create a new list item element
-  const skill = document.createElement('li');
+for (let skill of skills) {
+ let li = document.createElement('li');
+ li.innerText = skill;
+ skillsList.appendChild(li);
+}
+let messageForm = document.querySelector("[name='leave_message']");
+let messageSection = document.getElementById('messages');
+let messageList = messageSection.querySelector('ul');
+messageSection.hidden = true;
+let idCounter = 0;
+function makeId() {
+	let id = 'entry' + idCounter++;
+	return id;
+}
 
-  // Set the inner text of the list item element to the current skill
-  skill.innerText = skills[i];
+let entryById={};
+messageForm.addEventListener('submit', (event) => {
+	event.preventDefault();
+	let name = event.target.usersName.value;
+	let email = event.target.usersEmail.value;
+	let message = event.target.usersMessage.value;
+
+	console.log('Name:', name);
+	console.log('Email:', email);
+	console.log('Message:', message);
+	const uid = makeId();
+	let newMessage = document.createElement('li');
+
+	newMessage.innerHTML = `<a  href="mailto:${email} ">${name}  </a><span>wrote: ${message} </span>`;
+	newMessage.setAttribute('id', uid);
+
+	entryById[uid] = {usersName: name, usersEmail: email, usersMessage:message};
+	newMessage.appendChild(makeRemoveButton(uid));
+	newMessage.appendChild(makeEditButton(uid));
+
+	messageList.appendChild(newMessage);
+	messageForm.reset();
+	messageSection.style.display = 'block';
+	messageSection.hidden = false;
+});
+
+function makeRemoveButton(uid){
+	let removeButton = document.createElement('button');
+	removeButton.innerText = 'remove';
+	removeButton.type = 'button';
+	removeButton.addEventListener('click',  () => {
+		let entry = removeButton.parentNode;
+		let uid1 = entry.getAttribute('id');
+		delete entryById[uid1];
+		entry.remove();
+		if (messageList.childElementCount === 0){
+			messageSection.style.display = 'none';
+		};
+	});
+	return removeButton;
+};
+
+function makeEditButton(uid) {
+	const editButton = document.createElement('button');
+	editButton.innerText = 'edit';
+	editButton.type = 'button';
+	editButton.addEventListener('click', () => {
+		const entry = editButton.parentNode;
+		let clonedForm = messageForm.cloneNode(true);
+		clonedForm.usersName.value = entryById[uid].usersName;
+		clonedForm.usersEmail.value = entryById[uid].usersEmail;
+		clonedForm.usersMessage.value = entryById[uid].usersMessage;
+		clonedForm.addEventListener('submit', function editMessage(event) {
+			event.preventDefault();
+			entryById[uid].usersName = event.target.usersName.value;
+			entryById[uid].usersEmail = event.target.usersEmail.value;
+			entryById[uid].usersMessage = event.target.usersMessage.value;
+			entry.innerHTML = `<a href="mailto:${entryById[uid].usersEmail} "> ${entryById[uid].usersName} </a><span>wrote: ${entryById[uid].usersMessage}</span>`;
+			entry.appendChild(makeRemoveButton(uid));
+			entry.appendChild(makeEditButton(uid));
+			clonedForm.remove();
```

Comment:
clonedFrom.remove() is not necessary.  The cloneForm is not appended to the new entry so it is no longer there anyway.

Reply:
removed